### PR TITLE
New px4-dev-base and px4-dev-nuttx

### PR DESCRIPTION
**Pull request by dagar**
@AndreasAntener please review. I tried to get base as small and simple as possible. The bare minimum we need to build posix_sitl_default.

I have most of the others working, but to play it safe I'll pull them in one at a time, tagging and updating builds as I go. I pulled all the packages into the dockerfile (sorted) instead of your scripts because it works better with the layer caching.

Implements https://github.com/PX4/containers/issues/33

### CODE REVIEW

**Comment by AndreasAntener** [comment]
@dagar cool you're taking care of this :)
On the `RUN` commands: it is best practice to not add too many layers as it slows down file system access while running the containers (and there's an absolute layer limit I believe). Cache-wise the scripts should behave the same as the RUN commands (`ADD` should be cached as long as the files don't change). Did you notice problems?

File: docker/px4-dev/scripts/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Add local user
+# Either use the LOCAL_USER_ID if passed in at runtime or
+# fallback
+
+USER_ID=${LOCAL_USER_ID:-9001}
+
+echo "Starting with UID : $USER_ID"
+useradd --shell /bin/bash -u $USER_ID -o -c "" -m user
```
Comment (AndreasAntener) [inline docker/px4-dev/scripts/entrypoint.sh]: Did you check if this works when the user already exists? (entering the container a second time)

Comment (dagar) [inline docker/px4-dev/scripts/entrypoint.sh]: Yes, I'm able to enter the same container multiple times and it doesn't create a new user.

**Comment by dagar** [comment]
I thought the external script might be a problem with the cache, but it seems docker is pretty clever. Unless we share the script in multiple places I'd rather keep everything contained to one dockerfile.

If you want a script for installing deps what do you think about keeping it in Firmware/Tools and in sync with the dev guide?

Understood about multiple layers. That's why the snappy image ended up so large. Files added, installed, and deleted in multiple steps, which ended up keeping everything.

**Comment by AndreasAntener** [comment]
If you can get the SITL/ROS deps into good looking RUN commands I'm not against removing the scripts :). I think linking them from the guide is a good idea until you have specific things in there for Docker (or the other way around) and then it gets annoying, so I wouldn't worry about it.

**Comment by dagar** [comment]
@AndreasAntener okay with merging this?

**Comment by AndreasAntener** [comment]
Sure

**Pull request merged**